docs/spec.md

```diff
@@ -5377,20 +5391,24 @@ where `result_index[d] = operand_index[permutation[d]]`.
 
 | Label | Name          | Type                                         | Constraints |
 |-------|---------------|----------------------------------------------|-------------|
-| (I1)  | `operand`     | tensor or per-tensor quantized tensor        | (C1-C3)     |
-| (I2)  | `permutation` | 1-dimensional tensor constant of type `si64` | (C2), (C3)  |
+| (I1)  | `operand`     | tensor or per-tensor quantized tensor        | (C1-C5)     |
+| (I2)  | `permutation` | 1-dimensional tensor constant of type `si64` | (C1-C2)     |
 
 #### Outputs
 
 | Name     | Type                                  | Constraints |
 |----------|---------------------------------------|-------------|
-| `result` | tensor or per-tensor quantized tensor | (C1), (C3)  |
+| `result` | tensor or per-tensor quantized tensor | (C2-C5)     |
 
 #### Constraints
 
-* (C1) `element_type(operand) = element_type(result)`.
-* (C2) `permutation` is a permutation of `range(rank(operand))`.
-* (C3) `shape(result) = dim(operand, permutation...)`.
+* (C1) `permutation` is a permutation of `range(rank(operand))`.
+* (C2) `shape(result) = dim(operand, permutation...)`.
+* If the operation uses non-quantized tensors:
+  * (C3) `element_type(operand) = element_type(result)`.
+* If the operation uses quantized tensors:
+  * (C4) `is_quantized_tensor(operand) and is_quantized_tensor(result)`.
```

Comment:
For tanh operator, TFLite quantization prefers having a fixed quantization parameter range, instead of allowing the arbitrary quantization range. I wonder that kind of the restriction makes sense to the StableHLO quantization spec.

Reply:
Thanks @abattery for pointing the observation.  

I agree that StableHLO spec, in its current form, does not have the restriction ` (scale, zero_point) = (1.0 / 128.0, 0)`. The decision was deliberate for the initial version of the spec to be "open" preferring generality  and symmetry of IR design. On a related note, we have a open ticket https://github.com/openxla/stablehlo/issues/1405 to discuss similar issue. Also, updated the description of the issue to capture your point. Please let me know if it makes sense.